benchmarks/Kestrel.Performance/AsciiBytesToStringBenchmark.cs

```diff
@@ -0,0 +1,503 @@
+                            output[3] = (char)input[3];
+                            output[4] = (char)input[4];
+                            output[5] = (char)input[5];
+                            output[6] = (char)input[6];
+                            output[7] = (char)input[7];
+
+                            input += sizeof(long);
+                            output += sizeof(long);
+                        }
+                        if (input <= end - sizeof(int))
+                        {
+                            isValid &= CheckBytesInAsciiRange(((int*)input)[0]);
+
+                            output[0] = (char)input[0];
+                            output[1] = (char)input[1];
+                            output[2] = (char)input[2];
+                            output[3] = (char)input[3];
+
+                            input += sizeof(int);
+                            output += sizeof(int);
+                        }
+                    }
+                    else
+                    {
+                        // 32-bit: Loop ints by default
+                        while (input <= end - sizeof(int))
+                        {
+                            isValid &= CheckBytesInAsciiRange(((int*)input)[0]);
+
+                            output[0] = (char)input[0];
+                            output[1] = (char)input[1];
+                            output[2] = (char)input[2];
+                            output[3] = (char)input[3];
+
+                            input += sizeof(int);
+                            output += sizeof(int);
+                        }
+                    }
+                    if (input <= end - sizeof(short))
+                    {
+                        isValid &= CheckBytesInAsciiRange(((short*)input)[0]);
+
+                        output[0] = (char)input[0];
+                        output[1] = (char)input[1];
+
+                        input += sizeof(short);
+                        output += sizeof(short);
+                    }
+                    if (input < end)
+                    {
+                        isValid &= CheckBytesInAsciiRange(((sbyte*)input)[0]);
+                        output[0] = (char)input[0];
+                    }
+
+                    return isValid;
+                }
+
+                // do/while as entry condition already checked
+                do
+                {
+                    var vector = Unsafe.AsRef<Vector<sbyte>>(input);
+                    isValid &= CheckBytesInAsciiRange(vector);
+                    // Vectorized widen, byte vector to two short vectors
+                    // Split to two stage https://github.com/dotnet/coreclr/issues/13024#issuecomment-321413029
+                    Vector.Widen(vector, out var vector1, out var vector2);
+                    Unsafe.AsRef<Vector<short>>(output) = vector1;
+                    Unsafe.AsRef<Vector<short>>(output + Vector<short>.Count) = vector2;
```

Comment:
Nice! Reads much better than loop unrolling.

Since https://github.com/dotnet/coreclr/issues/13568 was already merged, should we wait for that before merging this?

Reply:
Was merged a while ago; so should be fine, I just didn't check the asm to confirm it produced a good output.

Suppose I should measure both :)

Reply:
Been working on too many simultaneous PRs 😆